### PR TITLE
Development/ga/ogd 86 update existing api edit user

### PR DESCRIPTION
**Pull request by ben-james-hmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/GA-86

### Change description ###

Branch being commited early as changes are blocking another ticket:

Added method to update user within manage user component.
Added new model and replaced previous user model in both manage user and invite user process to make both strongly typed.
Added several tests for new update user method
Added all required subscribes.

Outstanding tasks:

Add new component for update user failure with link back to relevant screen.
Add addtional tests to check subscribes (error, success, service down)

### CODE REVIEW

File: src/users/containers/manage-user/manage-user.component.spec.ts [hunk 1]
```diff
@@ -9,20 +9,25 @@ import * as fromRoot from '../../../app/store';
...
 import { LoggerService } from 'src/shared/services/logger.service';
 import { OrganisationDetails } from 'src/models';
+import { EditUserModel } from 'src/user-profile/models/editUser.model';
 import { RpxTranslatePipe, RpxTranslationService } from 'rpx-xui-translation';
+import { StandardUserPermissionsComponent } from 'src/users/components/standard-user-permissions/standard-user-permissions.component';
+import { UserPersonalDetailsComponent } from 'src/users/components/user-personal-details/user-personal-details.component';
+import { AsyncPipe } from '@angular/common';
 
-describe('ManageUserComponent', () => {
+fdescribe('ManageUserComponent', () => {
```
Comment (shaed-parkar) [inline src/users/containers/manage-user/manage-user.component.spec.ts:22]: remove the f here else all the other tests won't run

File: src/users/containers/utils/user-roles-util.ts [hunk 2]
```diff
@@ -36,14 +38,16 @@ export class UserRolesUtil {
     return roles;
   }
 
-  public static mapEditUserRoles(user: any, rolesAdd: any[], rolesDelete: any[]): any {
+  public static mapEditUserRoles(user: any, userId: string, rolesAdd: any[], rolesDelete: any[], accessTypes: UserAccessType[] = []): EditUserModel {
```
Comment (shaed-parkar) [inline src/users/containers/utils/user-roles-util.ts:41]: do you still need user id if the user object is being provided?

File: src/user-profile/services/user.service.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ export class UserService {
   constructor(private readonly http: HttpClient) {}
 
   public editUserPermissions(editUser): Observable<any> {
```
[the remark this answers is not in the document]
Comment (ben-james-hmcts) [reply src/user-profile/services/user.service.ts]: type added as specified

File: src/users/containers/manage-user/manage-user.component.spec.ts (hunk 1 above)
Comment (ben-james-hmcts) [reply src/users/containers/manage-user/manage-user.component.spec.ts]: f removed to allow all tests to run

File: src/users/containers/utils/user-roles-util.ts (hunk 2 above)
Comment (ben-james-hmcts) [reply src/users/containers/utils/user-roles-util.ts]: userId param removed. Tested both new and old edit user functions and both work using the id within the user object

**Review by shaed-parkar — APPROVED** [review]

**Pull request merged**